### DISCUSSION TITLE
Guidence on enabling wireless for all advanced turtles.

### DISCUSSION THREAD

**Discussion by Visc0sity** [cc-tweaked/CC-Tweaked#1916] [Q&A] [answered]
Hello, I hope you are all doing well! I want to start by clerifying that I am not asking anyone to make the change for me but if anyone could point me in the right direction i would greatly appreciate it.

I have always felt that the advanced turtle's color screen and max fuel boost doesnt add much. I would like to make a fork that would allow me to change the recipie of the advanced turtle to require an ender modem and then enable wireless on all advanced turtles, even without a modem attached.

I was looking through the files but with 2.5k entries for turtle and 1.5k entries for modem im not terrably sure where to start to find the right files or what form they would come in( a single file that just changes what api the computers can use, or having to effectively add all the modem code to the advanced turtle, etc) 

Thank you for taking the time to read this, and again, im not asking anyone to make this, but any assistance would mean the world.

**Comment by SquidDev** [comment, marked as answer]
So I *think* this should be fairly easy to hack in, though I haven't actually tested this:

In [`TurtleBlockEntity.createComputer`](https://github.com/cc-tweaked/CC-Tweaked/blob/cbe075b001c46cee95e00b8151f59f86203a03ac/projects/common/src/main/java/dan200/computercraft/shared/turtle/blocks/TurtleBlockEntity.java#L74-L84), add the following code:

```diff
  computer.addAPI(new TurtleAPI(computer, brain));
  brain.setupComputer(computer);
+ // Attach a modem to the back of the computer. This is a massive hack, but the easiest way to do it without duplicating code.
+ computer.setPeripheral(ComputerSide.BACK, new TurtleModem(new ResoureLocation("computercraft", "fake_modem"), ItemStack.EMPTY, true).createPeripheral(brain, TurtleSide.LEFT));
  return computer;
```

However, the modem will probably be auto-removed next time the turtle moves. You can prevent this by modifying [`TurtleBlockEntity.isPeripheralBlockedOnSide`](https://github.com/cc-tweaked/CC-Tweaked/blob/cbe075b001c46cee95e00b8151f59f86203a03ac/projects/common/src/main/java/dan200/computercraft/shared/turtle/blocks/TurtleBlockEntity.java#L151C1-L155C6):

```java
@Override
protected boolean isPeripheralBlockedOnSide(ComputerSide localSide) {
    return localSide == ComputerSide.BACK || hasPeripheralUpgradeOnSide(localSide);
}
```

**Reply by Visc0sity** [reply]
Thank you for your advice! So far ive run into issues building even without changing the code due to fabric test issues but I was able to assemble and run forge. Im trying the changes and ran into some errors. Both are invalid symbol, the first regarding turtles was fixed by importing 
`dan200.computercraft.shared.turtle.upgrades.TurtleModem;`

But the other error I dont understand quite yet:
```
CC-Tweaked\projects\common\src\main\java\dan200\computercraft\shared\turtle\blocks\TurtleBlockEntity.java:85: error: cannot find symbol
        computer.setPeripheral(ComputerSide.BACK, (new TurtleModem(new ResoureLocation("computercraft", "fake_modem"), ItemStack.EMPTY, true).createPeripheral(brain, TurtleSide.LEFT)));
                                                                       
  symbol:   class ResoureLocation
  location: class TurtleBlockEntity
```
'What is ResoureLocation? I didnt find any other refrences to it when searching the mod files. ignore that part i just saw some refrences to it, though im still looking'
Dont you love when your brain skips 90% of a word. it assembles wonderfully after realizing that resoure isnt a word 🤦

**Reply by SquidDev** [reply]
Oh, that should be `ResourceLocation`, sorry!

**Reply by Visc0sity** [reply]
You are good, and the code works! (even though i spent 20 minutes wondering why it wouldnt move when i didnt give it fuel) I greatly appreciate it! Once i fugure out more of grandle and package this I should be golden!
